packages/flutter/lib/src/material/data_table.dart

```diff
@@ -542,15 +540,8 @@ class DataTable extends StatelessWidget {
       height: headingRowHeight,
       alignment: numeric ? Alignment.centerRight : AlignmentDirectional.centerStart,
       child: AnimatedDefaultTextStyle(
-        style: TextStyle(
-          // TODO(ianh): font family should match Theme; see https://github.com/flutter/flutter/issues/3116
-          fontWeight: FontWeight.w500,
-          fontSize: _headingFontSize,
-          height: math.min(1.0, headingRowHeight / _headingFontSize),
-          color: (Theme.of(context).brightness == Brightness.light)
-            ? ((onSort != null && sorted) ? Colors.black87 : Colors.black54)
-            : ((onSort != null && sorted) ? Colors.white : Colors.white70),
-        ),
+        // Is this correct/breaking?
+        style: Theme.of(context).textTheme.headline6,
```

Comment:
@HansMuller is this what the todo is implying here? Would this be appropriate to match spec?